tests/object_sync_test.go

```diff
@@ -1049,6 +1213,685 @@ func makeTestNotifications(t *testing.T) []Notification {
+			// verify that these IDs represent those of the children hosts of this group.
+			if _, ok := checkableChildNodeIds[node.FromNodeId.String()]; !ok {
+				checkableChildNodeIds[node.FromNodeId.String()] = make(map[string]bool)
+			}
+			checkableChildNodeIds[node.FromNodeId.String()][node.ToNodeId.String()] = true
+		}
+	}
+	assert.NoError(t, dbParentNodes.Err(), "scanned parent node rows should not have errors")
+
+	expectedParentCount := len(totalExpectedParents)
+	if g.IsRedundancyGroup() {
+		expectedParentCount = 1 // All the children should have the redundancy group as parent node!
+		assert.Lenf(t, redundancyGroupChildNodeIds, len(totalExpectedChildren), "all children %v should have the redundancy group as parent node", totalExpectedChildren)
+		for _, child := range totalExpectedChildren {
+			h := checkables[child]
+			require.NotNil(t, h, "child node should be a Checkable")
+			assert.Truef(t, redundancyGroupChildNodeIds[h.NodeId.String()], "child node %q should have the redundancy group as parent node", child)
+			// The edge state ID of all the children of this redundancy group should be the same as the ID
+			// of the redundancy group ID, i.e. they all share the same edge state. This is just a duplicate check!
+			assert.Equalf(t, redundancyGroup.Id, h.EdgeStateId, "child node %q should have the correct edge state", child)
+		}
+	} else {
+		for _, child := range totalExpectedChildren {
+			h := checkables[child]
+			require.NotNilf(t, h, "child node %q should be a Checkable", child)
+
+			parents := checkableChildNodeIds[h.NodeId.String()]
+			require.NotNilf(t, parents, "child node %q should have parent nodes", child)
+			assert.Lenf(t, parents, len(totalExpectedParents), "child node %q should reference %d parent nodes", child, len(totalExpectedParents))
+
+			// Verify that the parent nodes of the children hosts are the correct ones.
+			for _, p := range totalExpectedParents {
+				parent := checkables[p]
+				require.NotNilf(t, parent, "parent node %q should be an existing Checkable", p)
+				assert.Truef(t, parents[parent.NodeId.String()], "child node %q should reference parent node %q", child, parent)
+			}
+		}
+	}
+	assert.Len(t, parentNodes, expectedParentCount)
+
+	for _, node := range parentNodes {
+		assert.NotNilf(t, node.FromNodeId, "parent node %q should have a from_node_id set", node.Name)
+		assert.NotNilf(t, node.ToNodeId, "parent node %q should have a to_node_id set", node.Name)
+
+		if g.IsRedundancyGroup() {
+			assert.Equal(t, node.Name, g.Name, "parent node should be the redundancy group itself")
+			assert.Equal(t, redundancyGroup.Id, node.RedundancyGroupId, "parent node should reference the redundancy group")
+
+			// Verify whether the connection between the current redundancy group and the parent Checkable is correct.
+			query := `SELECT NULL AS node_id, '' AS name, from_node_id, to_node_id FROM dependency_edge WHERE from_node_id = ?`
```

Comment:
When scanning into a struct, fields that are not part of the result aren't set, so you don't need to select `NULL`/`''` here.